File: src/Entity/Avis.php
<?php

namespace App\Entity;
use Symfony\Component\Validator\Constraints as Assert;

use Doctrine\ORM\Mapping as ORM;

/**
 * Avis
 *
 * @ORM\Table(name="avis", indexes={@ORM\Index(name="fk_avis_client", columns={"idClient"}), @ORM\Index(name="fk_avis_produit", columns={"referenceProduit"})})
 * @ORM\Entity(repositoryClass="App\Repository\AvisRepository")
 *
 */
class Avis
{
    /**
     * @var int
     *
     * @ORM\Column(name="idAvis", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idavis;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="string", length=100, nullable=false)
     * @Assert\NotBlank(message="Champ obligatoire")
     * @Assert\Type("string",message="Veuillez saisir une chaine")
     */
    private $description;

    /**
     * @var string
     *
     * @ORM\Column(name="type", type="string", length=50, nullable=false)
     * @Assert\NotBlank(message="Champ obligatoire")
     */
    private $type;

    /**
     * @var \Client
     *
     * @ORM\ManyToOne(targetEntity="Client")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idClient", referencedColumnName="idClient")
     * })
     * @Assert\NotBlank(message="Champ obligatoire")

     */
    private $idclient;

    /**
     * @var \Produit
     *
     * @ORM\ManyToOne(targetEntity="Produit")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="referenceProduit", referencedColumnName="reference")
     * })
     * @Assert\NotBlank(message="Champ obligatoire")
     */
    private $referenceproduit;

    public function getIdavis(): ?int
    {
        return $this->idavis;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getIdclient(): ?Client
    {
        return $this->idclient;
    }

    public function setIdclient(?Client $idclient): self
    {
        $this->idclient = $idclient;

        return $this;
    }

    public function getReferenceproduit(): ?Produit
    {
        return $this->referenceproduit;
    }

    public function setReferenceproduit(?Produit $referenceproduit): self
    {
        $this->referenceproduit = $referenceproduit;

        return $this;
    }


}

File: GamerGeeksCommunity-Mobile/src/entities/Avis.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

/**
 *
 * @author Mr
 */
public class Avis {

    private int idAvis;
    private int referenceProduit;
    private int idClient;
    private String description;
    private String type;

    public Avis() {

    }

    public Avis(int referenceProduit,int idClient, String description, String type) {
        this.referenceProduit = referenceProduit;
        this.description = description;
        this.idClient=idClient;
        if (type.equals("excellent") || type.equals("mediocre") || type.equals("moyen")) {
            this.type = type;
        } else {
            System.out.println("le type doit etre excellent , moyen ou mediocre");
        }
    }

    public int getIdClient() {
        return idClient;
    }

    public void setIdClient(int idClient) {
        this.idClient = idClient;
    }

    public int getIdAvis() {
        return idAvis;
    }

    public int getReferenceProduit() {
        return referenceProduit;
    }

    public String getDescription() {
        return description;
    }

    public String getType() {
        return type;
    }

    public void setReferenceProduit(int referenceProduit) {
        this.referenceProduit = referenceProduit;
    }

    public void setIdAvis(int idAvis) {
        this.idAvis = idAvis;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setType(String type) {
        if (type.equals("excellent") || type.equals("mediocre") || type.equals("moyen")) {
            this.type = type;
        } else {
            System.out.println("le type doit etre excellent , moyen ou mediocre");
        }
    }

    @Override
    public String toString() {
        return "Avis{" + "idAvis=" + idAvis + ", referenceProduit=" + referenceProduit + ", idClient=" + idClient + ", description=" + description + ", type=" + type + '}';
    }

    
}
